src/rest/connector/libs/bigip/implementation.py

```diff
@@ -81,79 +81,72 @@ def connected(self):
         return self._is_connected
 
     def disconnect(self):
-
         """disconnect the device for this particular alias"""
+        if self.token:
+            try:
+                log.info("Deleting token: '%s'", self.token)
+                delete_url = (
+                    f"{self.base_url}/mgmt/shared/authz/tokens/{self.token}"
+                )
```

Comment:
While I don't personally dislike this approach, this seems like a really good way to break legacy usage. Anyone using this with `self.ip` and `self.port` instead of `self.base_url` is going to have a bad time. 

I'd say you should probably do something like this
```suggestion
                if self.base_url:
                    delete_url = (
                        f"{self.base_url}/mgmt/shared/authz/tokens/{self.token}"
                    )
                else:
                    delete_url = "https://{0}:{1}/mgmt/shared/authz/tokens/{2}".format(
                        self.ip, self.port, self.token
                    )
```

Reply:
Nevermind. I see later on `self.base_url` is made up of `self.ip` and `self.port`. Leaving the comment for you to confirm this

Reply:
Yes, that's exactly the case.
The API is fully backwards compatible, just been extended with some new things, like retries